### PR TITLE
cryptomator: 1.5.15 -> 1.6.4

### PR DESCRIPTION
**Pull request by peterwilli**
###### Motivation for this change

Cryptomator was out of date. It was also broken, and needed fixing (see: https://github.com/NixOS/nixpkgs/issues/150391)

###### Things done

I fixed the bug by changing from OpenJDK to Zulu 17. For this, I also had to make a package for Zulu 17 as it didn't exist yet. So that too is added with this PR.

I was awake a few nights for this peculiarity, I feel I'm experienced enough now to maintain this package so I added myself as maintainer in addition to bachp.

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by peterwilli** [comment]
I don't own a Mac. If someone can test the Darwin version of Zulu17 that'd be lovely!

**Comment by bakhtiyarneyman** [comment]
Thanks for doing this Peter! From looking at the failure it seems that you need to add yourself to the `maintainer-list.nix`?

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -13234,6 +13234,7 @@ with pkgs;
 
   zulip-term = callPackage ../applications/networking/instant-messengers/zulip-term { };
 
+  zulu17 = callPackage ../development/compilers/zulu/17.nix { };
   zulu8 = callPackage ../development/compilers/zulu/8.nix { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:13238]:
```suggestion
  zulu8 = callPackage ../development/compilers/zulu/8.nix { };
  zulu17 = callPackage ../development/compilers/zulu/17.nix { };
```

File: pkgs/tools/security/cryptomator/default.nix [hunk 2]
```diff
@@ -44,48 +45,50 @@ let
...
-    mkdir -p $out/bin/ $out/usr/share/cryptomator/libs/
-
-    cp buildkit/target/libs/* buildkit/target/linux-libs/* $out/usr/share/cryptomator/libs/
-
-    makeWrapper ${jre}/bin/java $out/bin/cryptomator \
-      --add-flags "-classpath '$out/usr/share/cryptomator/libs/*'" \
+    mkdir -p $out/bin $out/libs $out/mods
+    # moving over all libs
+    cp -r target/mods/* $out/mods
+    cp target/cryptomator-1.6.4.jar $out/mods
```
Comment (SuperSandro2000) [inline pkgs/tools/security/cryptomator/default.nix:62]: Please use ${version} or a wildcard.

**Comment by bachp** [comment]
@peterwilli I just saw this PR after creating https://github.com/NixOS/nixpkgs/pull/160628

I took a different route to fix the issue by just replacing jffi not the whole jdk.

As for the openjfx issue I added version 17 to nixpkgs https://github.com/NixOS/nixpkgs/pull/155746 so the hack is no longer necessary.

Other than that I think we came to pretty much the same changes required to for 1.6.x  :wink: 

Do you mind if we close this MR? I can add you as an additional maintainer my MR if that is OK with you.

**Comment by peterwilli** [comment]
> @peterwilli I just saw this PR after creating #160628
> 
> I took a different route to fix the issue by just replacing jffi not the whole jdk.
> 
> As for the openjfx issue I added version 17 to nixpkgs #155746 so the hack is no longer necessary.
> 
> Other than that I think we came to pretty much the same changes required to for 1.6.x wink
> 
> Do you mind if we close this MR? I can add you as an additional maintainer my MR if that is OK with you.

Hey bachp. Great minds think alike! That's alright with me. I'm going to check your PR just to see what you did differently, would be fun to learn from!

**Pull request closed without merging**